### PR TITLE
feat: :sparkles: forgejo support added

### LINKED ISSUE #14: Gitea support
### Describe the feature or enhancement

Would very beneficial to add support for [gitea](https://about.gitea.com/) as well.

### PR DESCRIPTION
**Pull request by jbonadiman**
This pull request adds support for Forgejo through the `ListMyRepos` SDK method, without special treatment for orgs, since this would bring a little bit of complexity due to orgs being handled as any other user account in the platform.
But the include/exclude config would still work, if the repos are listed in the config as `<org-name>/<repo-name>`.

The Forgejo client needs the an access token with read permissions to `user` and `repository` and it can be created in [this page](https://codeberg.org/user/settings/applications).
It also needs a server url, and for that I used the `server.domain` and `server.protocol` configs.

It might close #14, but I don't have an account on Gitea to test.

### CODE REVIEW

**Comment by AkashRajpurohit** [comment]
Thanks for opening this @jbonadiman, I'll try to take a look at this over the weekend.
To confirm few things, you were able to test and verify that it works with forgejo account right?

You did not test it for Gitea though but have a hunch that it should work for Gitea as well?

**Comment by jbonadiman** [comment]
> Thanks for opening this @jbonadiman, I'll try to take a look at this over the weekend. To confirm few things, you were able to test and verify that it works with forgejo account right?
> 
> You did not test it for Gitea though but have a hunch that it should work for Gitea as well?

Yes, that's right. I tested it in my Forgejo account with a normal repo and an org repo and it worked.
And I didn't test for Gitea.

File: cmd/root.go [hunk 1]
```diff
@@ -89,6 +90,10 @@ var rootCmd = &cobra.Command{
 			client = gitlab.NewGitlabClient(cfg.Token)
 		case "bitbucket":
 			client = bitbucket.NewBitbucketClient(cfg.Username, cfg.Token)
+		case "forgejo":
+			client = forgejo.NewForgejoClient(
+				fmt.Sprintf("%s://%s", cfg.Server.Protocol, cfg.Server.Domain),
```
Comment (AkashRajpurohit) [inline cmd/root.go:95]: Can we pass the cfg.Server to the new client function and handling the formatting of the base URL there?

**Review by AkashRajpurohit — APPROVED** [review]
Just a small change but overall the PR looks good to me. Thanks for this @jbonadiman

**Comment by AkashRajpurohit** [comment]
One additional note: since it's not been tested that the setup for forgejo works out of the box for gitea as well, I would not mark the existing issue of #14 support for Gitea as closed as part of the release.

Post release if the user verifies that it works as it is, we can then decide on merging these two clients and mention the same.

File: cmd/root.go (hunk 1 above)
Comment (jbonadiman) [reply cmd/root.go:95]: Yes, sure, that makes sense. I'll push a commit right away.

**Comment by AkashRajpurohit** [comment]
Thanks for adding this @jbonadiman 🎉

**Pull request merged**